checkup.go

```diff
@@ -359,6 +359,12 @@ type Storage interface {
 	Store([]Result) error
 }
 
+// StorageReader can read results from the Storage.
+type StorageReader interface {
+	Fetch(string) ([]Result, error)
+	GetIndex() (map[string]int64, error)
```

Comment:
On second thought, having the index (key->timestamp) is useful, and the s3 backend seems not to use an index at all anyway, so I don't think the interface should be changed.

I'll document the methods as suggested.